src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -192,20 +195,47 @@ def enable_sequential_cpu_offload(self, gpu_id=0):
 
         device = torch.device(f"cuda:{gpu_id}")
 
-        for cpu_offloaded_model in [self.unet, self.text_encoder, self.vae]:
+        for cpu_offloaded_model in [self.unet, self.text_encoder, self.decoder]:
             cpu_offload(cpu_offloaded_model, device)
 
         if self.safety_checker is not None:
             cpu_offload(self.safety_checker, execution_device=device, offload_buffers=True)
 
+    def enable_model_offload(self, gpu_id=0):
+        r"""
+        Offloads all models to CPU using accelerate, reducing memory usage with a low impact on
+        performance. Compared to `enable_sequential_cpu_offload`, this method moves one model
+        at a time to the GPU when its `forward` method is called, and the model remains in GPU
+        until the next model runs. Memory savings are lower than with `enable_sequential_cpu_offload`,
+        but performance is much better due to the iterative execution of the `unet`.
+        """
+        if is_accelerate_available() and is_accelerate_version(">=", "0.17.0.dev0"):
+            from accelerate import cpu_offload_with_hook
+        else:
+            raise ImportError("`enable_model_offload` requires `accelerate v0.17.0` or higher.")
+
+        device = torch.device(f"cuda:{gpu_id}")
+
+        hook = None
+        for cpu_offloaded_model in [self.text_encoder, self.unet, self.decoder]:
+            _, hook = cpu_offload_with_hook(cpu_offloaded_model, device, prev_module_hook=hook)
+
+        if self.safety_checker is not None:
+            _, hook = cpu_offload_with_hook(self.safety_checker, device, prev_module_hook=hook)
+
+        # We'll offload the last model manually.
+        # Alternative: daisy-chain with the first one so a new inference will offload the last model,
+        # but then we need access to `hook.hook.prev_module_hook`.
+        self.offload_hook = hook
+
     @property
     def _execution_device(self):
         r"""
         Returns the device on which the pipeline's models will be executed. After calling
         `pipeline.enable_sequential_cpu_offload()` the execution device can only be inferred from Accelerate's module
         hooks.
         """
-        if self.device != torch.device("meta") or not hasattr(self.unet, "_hf_hook"):
+        if not hasattr(self.unet, "_hf_hook"):
```

Comment:
Device is no longer `meta` when using `enable_model_offload`. I'm not fully sure why we need it for `enable_sequential_cpu_offload`, as `_hf_hook` seems to be enough (testing neeeded).

Reply:
LGTM